### PR TITLE
[IAST] Check status != 404 and enable nodejs tests

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
Check for endopint response != 404
Enable test for node: "Strong hash algorithm is not reported as insecure"

### CODE REVIEW

File: tests/appsec/iast/test_iast.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def test_insecure_hash_multiple(self):
             r, vulnerability_count=2, type="WEAK_HASH", location_path=self.EXPECTED_LOCATION
         )
 
-    @missing_feature(library="nodejs", reason="Need to be implement global vulnerability deduplication")
+    @missing_feature(context.library < "nodejs@3.3.1", reason="Need to be implement global vulnerability deduplication")
```
Comment (robertomonteromiguel) [inline tests/appsec/iast/test_iast.py:48]: @uurien paternity leave
@CarlesDD PTO until 27th
I think the commit that fix this test is merged in master: https://github.com/DataDog/dd-trace-js/commit/a721792b0eebfcf1a42422730b7b331e75f26618

File: tests/appsec/iast/test_iast.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 @released(
     dotnet="?",
     golang="?",
-    java={"spring-boot": "0.108.0", "*": "?"},
+    java={"spring-boot": "0.108.0", "spring-boot-jetty": "0.109.0", "*": "?"},
```
Comment (robertomonteromiguel) [inline tests/appsec/iast/test_iast.py:12]: The remain fixes for Java will be added from this PR: https://github.com/DataDog/system-tests/pull/508

File: utils/interfaces/_library/appsec_iast.py [hunk 3]
```diff
@@ -34,6 +35,8 @@ def final_check(self):
         def vulnerability_dict(vulDict):
             return namedtuple("X", vulDict.keys())(*vulDict.values())
 
+        if self.request.status_code == 404:
+            self.set_failure(f"Called endpoint wasn't available. Status code: 404")
```
Comment (robertomonteromiguel) [inline utils/interfaces/_library/appsec_iast.py:39]: we will always fail if the endpoint doesn't exist

File: utils/interfaces/_library/appsec_iast.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class _BaseAppSecIastValidation(BaseValidation):
     def __init__(self, request):
         super().__init__(request=request)
         self.appsec_iast_events = []  # list of all appsec iast events
+        self.request = request
```
Comment (robertomonteromiguel) [inline utils/interfaces/_library/appsec_iast.py:22]: I called request like in another python files, but, indeed, wouldn't it be a response?

Comment (cbeauchesne) [reply utils/interfaces/_library/appsec_iast.py:22]: You are perfectly right ...

Though, it'll be for another PR, pretty big one indeed ...

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**